homeassistant/components/filesize/coordinator.py

```diff
@@ -26,15 +27,34 @@ def __init__(self, hass: HomeAssistant, path: str) -> None:
             update_interval=timedelta(seconds=60),
             always_update=False,
         )
-        self._path = path
+        self._unresolved_path = unresolved_path
+        self._path: pathlib.Path | None = None
 
-    async def _async_update_data(self) -> dict[str, float | int | datetime]:
+    def _get_full_path(self) -> pathlib.Path:
+        """Check if path is valid, allowed and return full path."""
+        path = self._unresolved_path
+        get_path = pathlib.Path(path)
+        if not get_path.exists() or not get_path.is_file():
+            raise UpdateFailed(f"Can not access file {path}")
+
+        if not self.hass.config.is_allowed_path(path):
+            raise UpdateFailed(f"Filepath {path} is not valid or allowed")
```

Comment:
Let's move them around.

```suggestion
        if not self.hass.config.is_allowed_path(path):
            raise UpdateFailed(f"Filepath {path} is not valid or allowed")

        if not get_path.exists() or not get_path.is_file():
            raise UpdateFailed(f"Can not access file {path}")
```